File: LeetCode/2132-convert-1d-array-into-2d-array/convert-1d-array-into-2d-array.py
class Solution:
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        if len(original) != m*n:
            return []
        result = []
        
        for i in range(0, len(original), n):
            arr = []
            for j in range(i, i + n):
                arr.append(original[j])
            result.append(arr)
        return result

File: LeetCode/2132-convert-1d-array-into-2d-array/convert-1d-array-into-2d-array.cpp
class Solution {
public:
    vector<vector<int>> construct2DArray(vector<int>& original, int m, int n) {
        int len = original.size();
        if(len != m * n){
            return {};
        }

        vector<vector<int>> result;
        for(int i = 0; i < len; i += n){
            vector<int> arr;
            for(int j = i; j < i + n; j += 1){
                arr.push_back(original[j]);
            }

            result.push_back(arr);
        }

        return result;
    }
};